compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToROCDL.cpp

```diff
@@ -78,6 +78,221 @@ static void populateConvertGPUToAMDGPUPatterns(RewritePatternSet &patterns,
+                            LLVM::LLVMFunctionType::get(llvmI64, {llvmI64}));
+    LLVM::LLVMFuncOp ocklAppendArgs;
+    if (!adaptor.getArgs().empty()) {
+      ocklAppendArgs = getOrDefineFunction(
+          moduleOp, loc, rewriter, "__ockl_printf_append_args",
+          LLVM::LLVMFunctionType::get(llvmI64,
+                                      {llvmI64, /*numArgs*/ llvmI32, llvmI64,
+                                       llvmI64, llvmI64, llvmI64, llvmI64,
+                                       llvmI64, llvmI64, /*isLast*/ llvmI32}));
+    }
+    LLVM::LLVMFuncOp ocklAppendStringN = getOrDefineFunction(
+        moduleOp, loc, rewriter, "__ockl_printf_append_string_n",
+        LLVM::LLVMFunctionType::get(llvmI64, {llvmI64, ptrType,
+                                              /*length (bytes)*/ llvmI64,
+                                              /*isLast*/ llvmI32}));
+
+    // Start the printf hostcall.
+    Value zeroI64 = rewriter.create<LLVM::ConstantOp>(loc, llvmI64, 0);
+    auto printfBeginCall =
+        rewriter.create<LLVM::CallOp>(loc, ocklBegin, zeroI64);
+    Value printfDesc = printfBeginCall.getResult();
+
+    // Create the global op or find an existing one.
+    LLVM::GlobalOp global = getOrCreateStringConstant(
+        rewriter, loc, moduleOp, llvmI8, "printfFormat_", adaptor.getFormat());
+
+    // Get a pointer to the format string's first element and pass it to
+    // printf().
+    Value globalPtr = rewriter.create<LLVM::AddressOfOp>(
+        loc,
+        LLVM::LLVMPointerType::get(rewriter.getContext(),
+                                   global.getAddrSpace()),
+        global.getSymNameAttr());
+    Value stringStart =
+        rewriter.create<LLVM::GEPOp>(loc, ptrType, global.getGlobalType(),
+                                     globalPtr, ArrayRef<LLVM::GEPArg>{0, 0});
+    Value stringLen = rewriter.create<LLVM::ConstantOp>(
+        loc, llvmI64, cast<StringAttr>(global.getValueAttr()).size());
+
+    Value oneI32 = rewriter.create<LLVM::ConstantOp>(loc, llvmI32, 1);
+    Value zeroI32 = rewriter.create<LLVM::ConstantOp>(loc, llvmI32, 0);
+
+    auto appendFormatCall = rewriter.create<LLVM::CallOp>(
+        loc, ocklAppendStringN,
+        ValueRange{printfDesc, stringStart, stringLen,
+                   adaptor.getArgs().empty() ? oneI32 : zeroI32});
+    printfDesc = appendFormatCall.getResult();
+
+    // __ockl_printf_append_args takes 7 values per append call
+    constexpr size_t argsPerAppend = 7;
+    size_t nArgs = adaptor.getArgs().size();
+    for (size_t group = 0; group < nArgs; group += argsPerAppend) {
+      size_t bound = std::min(group + argsPerAppend, nArgs);
+      size_t numArgsThisCall = bound - group;
+
+      SmallVector<Value, 2 + argsPerAppend + 1> arguments;
+      arguments.push_back(printfDesc);
+      arguments.push_back(
+          rewriter.create<LLVM::ConstantOp>(loc, llvmI32, numArgsThisCall));
+      for (Value arg : adaptor.getArgs()) {
```

Comment:
```suggestion
      for (Value arg : adaptor.getArgs().slice(group, numArgsThisCall)) {
```
This fails when you have > 7 arguments, unlike the upstream pattern. Here's the fix